torch/csrc/jit/ir.h

```diff
@@ -1104,15 +1103,19 @@ inline void Node::cloneFrom(Node * s) {
 	copyAttributes(*s);
 }
 
-inline Value* Value::setUniqueName(const std::string & name) {
-  if (name.find_first_not_of("0123456789") == std::string::npos) {
-    throw std::runtime_error("names may not be integers: " + name);
+inline Value* Value::setUniqueName(const std::string & orig_name) {
+  if (orig_name.find_first_not_of("0123456789") == std::string::npos) {
+    throw std::runtime_error("names may not be integers: " + orig_name);
   }
-  if (node_->graph_->unique_names_.find(name) != node_->graph_->unique_names_.end()) {
-    throw std::runtime_error("name is already in use in this graph: " + name);
+  auto & names = node()->owningGraph()->unique_names_;
+  auto name = orig_name;
+  for(size_t i = 1; names.find(name) != names.end(); i++) {
+    std::stringstream ss;
+    ss << orig_name << "." << i;
+    name = ss.str();
   }
-  node_->graph_->unique_names_.insert(name);
-  unique_name_ = name;
+  names.insert(name);
+  unique_name_ = std::move(name);
```

Comment:
How does this even build? What are `names`, and we do we do a linear scan now?

Reply:
Right -- I had to fix `setUniqueName`. It is used when nodes get copied, so it is not really reasonable for it to every throw an exception. It was breaking the ability for graph executor to optimize @script code. Instead, I took the LLVM behavior of giving the node a uniqueName that is close to the requested uniqueName (e.g. foo might become `foo.1` or `foo.2`). This is more robust while still preserving the debugging data. Names is just an alias for `node()->owningGraph()->unique_names_`. The linear scan only happens if there is a collision, so until we get into a situation where we have a lot of colliding names, this should perform fine.